modules/extensions-support/src/main/java/org/mule/module/extension/internal/manager/ConfigurationInstanceWrapper.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.manager;
+
+import static java.lang.String.format;
+import static org.mule.util.Preconditions.checkArgument;
+import org.mule.extension.introspection.Configuration;
+import org.mule.extension.introspection.Operation;
+import org.mule.extension.runtime.OperationExecutor;
+import org.mule.util.StringUtils;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * A wrapper class to hold an instance which is a realisation of a
+ * {@link Configuration} model and state associated to it, such as its
+ * {@link OperationExecutor}s, registration name, etc.
+ * <p/>
+ * This class is also useful to use a {@link #configurationInstance} in hash-based
```

Comment:
I could explain, but I won't because the need for this will go away after we change the OperationExecutor lifecycle. So just sit tight and wait for this to go away